### DISCUSSION TITLE
Explanation of Corner Plot Histograms

### DISCUSSION THREAD

**Discussion by Gaurav17Joshi** [Joshuaalbert/jaxns#100] [Q&A] [answered]
Hi, @Joshuaalbert , I wanted to understand the [Histograms](https://github.com/Joshuaalbert/jaxns/blob/main/jaxns/plotting.py#L69) made in the corner plot after the sampling run. The Histograms are not made of all the samples of the run. How and why are we taking selected weighted samples?

**Comment by Joshuaalbert** [comment, marked as answer]
Hi @Gaurav17Joshi the reason is because in nested sampling, the samples produced are not of equal weight, unlike MCMC. Instead they are weighted by how much they contribute to the evidence, `Z = sum_i dP_i`. Or, `dP_i = L_i * w_i`, where `w_i` is the amount of shrinkage between likelihood contours. These weights, `dP_i` allow you to take expectations, e.g. `E_posterior[f] = sum_i f(x_i) * dP_i`. When you have weights of equal weight, e.g. with MCMC, `dP_i=1`. Anyways, you can resample the weighted samples to produce equally weighted samples, by selecting a sample with replacement proportionally to `dP_i`. And that is what we do before doing any type of statistic that relies of equally weighted samples, such as the histogram. Hope that help! Let me know.

**Reply by Gaurav17Joshi** [reply]
Thanks for the explaination.
